controllers/vpcendpoint/helpers.go

```diff
@@ -79,12 +80,22 @@ func (r *VpcEndpointReconciler) parseClusterInfo(ctx context.Context, vpce *avov
 		r.log.V(1).Info("Parsed region from infrastructure", "region", region)
 	}
 
-	if refreshAWSSession {
-		cfg, err := config.LoadDefaultConfig(ctx, config.WithRegion(r.clusterInfo.region))
+	if vpce.Spec.AWSCredentialOverrideRef != nil {
+		// Use the provided override credentials for this specific vpcendpoint
+		cfg, err := secrets.ParseAWSCredentialOverride(ctx, r.Client, r.clusterInfo.region, vpce.Spec.AWSCredentialOverrideRef)
 		if err != nil {
 			return err
 		}
 		r.awsClient = aws_client.NewAwsClient(cfg)
+	} else {
```

Comment:
I'd suggest a unit test around this function to account for these new conditionals.

Reply:
Had a quick discussion, since we use the creds right after it's not straightforward to unit test as it is written and would need a bit of a refactor. Can focus on improving the testability of this in a future PR.